packages/http-specs/specs/type/union/discriminated/main.tsp

```diff
@@ -0,0 +1,199 @@
+import "@typespec/http";
+import "@typespec/spector";
+
+using Http;
+using Spector;
+
+/**
+ * Describe scenarios for discriminated unions.
+ */
+@scenarioService("/type/union/discriminated")
+namespace Type.Union.Discriminated;
+
+// Models for discriminated unions
+model Cat {
+  name: string;
+  meow: boolean;
+}
+
+model Dog {
+  name: string;
+  bark: boolean;
+}
+
+/**
+ * Test discriminated union with default envelope serialization.
+ * The discriminated union should serialize with "kind" as discriminator
+ * and "value" as envelope property.
+ */
+@discriminated
```

Comment:
is it allowed if Cat and Dog has "extend"? And what if they have property "kind" also?

```
model Pet {
    name: string;
}
model Cat {
  meow: boolean;
}

model Dog {
  bark: boolean;
}
```

Reply:
This PR is just to basic cases, and we could add more in the future if needed.